analyzer/evmtokenbalances/evm_token_balances.go

```diff
@@ -0,0 +1,303 @@
+package evmtokenbalances
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/iancoleman/strcase"
+	oasisConfig "github.com/oasisprotocol/oasis-sdk/client-sdk/go/config"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/oasisprotocol/oasis-indexer/analyzer"
+	"github.com/oasisprotocol/oasis-indexer/analyzer/modules"
+	"github.com/oasisprotocol/oasis-indexer/analyzer/util"
+	"github.com/oasisprotocol/oasis-indexer/config"
+	"github.com/oasisprotocol/oasis-indexer/log"
+	"github.com/oasisprotocol/oasis-indexer/storage"
+	"github.com/oasisprotocol/oasis-indexer/storage/oasis"
+)
+
+// Imagine a timeline starting from a `balanceOf` output `v0` followed by
+// a delta `d1` encountered in a `Transfer` log.
+// [v0][d1]
+//    ^   ^
+//    r0  r1
+// Consider two points in history:
+// - `r0` is the round that the token balances analyzer last downloaded the
+//   balance for.
+// - `r1` is a later round that the block scanner scanned where the balance
+//   changed.
+// Additionally, at this time:
+// - `last_download_round` is `r0`
+// - `last_mutate_round` is `r1`
+// - `balance` is `v0` + `d1`
+// - `extra` is some unimportant value, typically `d1`.
+//
+// Suppose at this point the token balances analyzer runs. It notices that
+// this account is _stale_, because `last_mutate_round` >
+// `last_download_round` on this account. It does the following atomically
+// (regardless of what the block scanner is doing):
+// 1. Set _next download round_ = `last_mutate_round` (= `r1`).
+// 2. Set `extra` = 0.
```

Comment:
turns out we don't need this column. can have token balances analyzer snapshot _old balance_ = `balance` = `v0` + `d1` when it starts. then later set `balance` = `v1` + `balance` - _old balance_ = `v1` + `d2`.

Reply:
snapshotted thusly